risc0/zkvm/src/host/api/convert.rs

```diff
@@ -1206,7 +1209,75 @@ impl TryFrom<pb::api::ProveKeccakRequest> for ProveKeccakRequest {
                 .control_root
                 .ok_or_else(|| malformed_err("ProveKeccakRequest.control_root"))?
                 .try_into()?,
-            input: value.input,
+            input,
         })
     }
 }
+
+pub(crate) fn keccak_input_to_bytes(input: &[KeccakState]) -> Vec<u8> {
+    // Note: safe to cast slice given alignment of KeccakState (8 bytes) is greater than
+    // the alignment of the u8 buffer.
+    bytemuck::cast_slice(input).to_vec()
+}
+
+pub(crate) fn try_keccak_bytes_to_input(input: &[u8]) -> Result<Vec<KeccakState>> {
+    let chunks = input.chunks_exact(std::mem::size_of::<KeccakState>());
+    if !chunks.remainder().is_empty() {
+        bail!("Input length must be a multiple of KeccakState size");
+    }
+    chunks
+        .map(bytemuck::try_pod_read_unaligned)
+        .collect::<Result<_, _>>()
+        .map_err(|e| anyhow!("Failed to convert input bytes to KeccakState: {}", e))
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_keccak_bytes_to_input_alignment() {
+        // Create a buffer with extra padding at the start to test different alignments
+        let padding = 8; // Test all possible alignments (0-7)
```

Comment:
Generally this can go up to 16 actually

Reply:
Fair enough, but in this case the only that matter are the u64 (8) based alignments?

Reply:
Well this is the alignment of the memory you are copying from, which can have an arbitrary alignment, technically it can have any kind of alignment but the ones that matter for real processors are (2, 4, 8, 16) IIUC. 

Thinking it through later, this test was probably just overkill, just testing one situation of a mismatched alignment would probably have been fine. Testing all the ways it can be unaligned is probably unneeded.